listeners.go

```diff
@@ -465,12 +465,70 @@ func ListenPacket(network, addr string) (net.PacketConn, error) {
 	return ln.(net.PacketConn), nil
 }
 
+// ListenQUIC returns a quic.EarlyListener suitable for use in a Caddy module.
+// The network will be transformed into a QUIC-compatible type (if unix, then
+// unixgram will be used; otherwise, udp will be used).
+//
+// NOTE: This API is EXPERIMENTAL and may be changed or removed.
+func (na NetworkAddress) ListenQUIC(ctx context.Context, portOffset uint, config net.ListenConfig, tlsConf *tls.Config, activeRequests *int64) (http3.QUICEarlyListener, error) {
```

Comment:
There's a lot of logic in here, as well as in the now-deprecated `caddy.ListenQUIC()`. Can we have `caddy.ListenQUIC()` call this one or would that be too much change do you think?

I just noticed the other ones (caddy.Listen, etc) don't call NetworkAddress.Listen(), so maybe those should too? I dunno. Just a thought. To help avoid duplicating code.

Reply:
`caddy.ListenQUIC()` takes a `net.PacketConn` and its life cycle is managed by the called. It doesn't feel right to call `NetworkAddress.ListenQUIC()`.

The others maybe changed to do so, but that will be another patch.

Reply:
Ok, duh, you're right. Don't worry about changing that for now. :+1: Thanks!